plugins/security/nexus-ldap-plugin/src/main/java/org/sonatype/security/realms/ldap/internal/persist/orient/LdapConfigurationEntityAdapter.java

```diff
@@ -0,0 +1,283 @@
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.security.realms.ldap.internal.persist.orient;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.orient.OClassNameBuilder;
+import org.sonatype.security.realms.ldap.internal.persist.entity.Connection;
+import org.sonatype.security.realms.ldap.internal.persist.entity.Connection.Host;
+import org.sonatype.security.realms.ldap.internal.persist.entity.Connection.Protocol;
+import org.sonatype.security.realms.ldap.internal.persist.entity.LdapConfiguration;
+import org.sonatype.security.realms.ldap.internal.persist.entity.Mapping;
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.metadata.schema.OClass;
+import com.orientechnologies.orient.core.metadata.schema.OClass.INDEX_TYPE;
+import com.orientechnologies.orient.core.metadata.schema.OSchema;
+import com.orientechnologies.orient.core.metadata.schema.OType;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * {@link LdapConfiguration} entity adapter.
+ *
+ * @since 3.0
+ */
+@Named
+@Singleton
+public class LdapConfigurationEntityAdapter
+    extends ComponentSupport
+{
+  private static final String DB_CLASS = new OClassNameBuilder()
+      .prefix("ldap")
+      .type(LdapConfiguration.class)
+      .build();
+
+  private static final String DB_CONNECTION_HOST_CLASS = new OClassNameBuilder()
+      .prefix("ldap")
+      .type(Host.class)
+      .build();
+
+  private static final String DB_CONNECTION_CLASS = new OClassNameBuilder()
+      .prefix("ldap")
+      .type(Connection.class)
+      .build();
+
+  private static final String DB_MAPPING_CLASS = new OClassNameBuilder()
+      .prefix("ldap")
+      .type(Mapping.class)
+      .build();
+
+  private static final String P_ID = "id";
+
+  private static final String P_NAME = "name";
+
+  private static final String P_ORDER = "order";
+
+  private static final String P_CONNECTION = "connection";
+
+  private static final String P_MAPPING = "mapping";
+
+  public OClass register(final ODatabaseDocumentTx db) {
+    checkNotNull(db);
+    OSchema schema = db.getMetadata().getSchema();
+    OClass type = schema.getClass(DB_CLASS);
+    if (type == null) {
+      // connection
+      OClass connectionHostType = schema.createClass(DB_CONNECTION_HOST_CLASS);
+      connectionHostType.createProperty("protocol", OType.STRING).setNotNull(true);
+      connectionHostType.createProperty("hostName", OType.STRING).setNotNull(true);
+      connectionHostType.createProperty("port", OType.INTEGER).setNotNull(true);
```

Comment:
Has the Nexus team started talking about standards in terms of schema with the move to OrientDB? 
In particular, should these be calling setMandatory(true) in addition to setNotNull()?
Some of these questions might be obviated by extending from here https://github.com/sonatype/nexus-oss/blob/NEXUS-7901_groovyUpgrade/components/nexus-component-services/src/main/java/org/sonatype/nexus/component/services/adapter/EntityAdapterSupport.java which provides helper methods for property creation that seems to enforce some standards.